concept/type/impl/AttributeTypeImpl.java

```diff
@@ -777,11 +776,13 @@ private String(GraphManager graphMgr, TypeVertex vertex) {
         @Override
         public void setRegex(Pattern regex) {
             if (regex != null) {
-                getInstances().parallel().forEach(attribute -> {
+                // TODO can we do this in parallel as it was before?
```

Comment:
question here: do we need to retain `parallel()` here? if so, we should examine how we can make our functional iterators have a parallel() method -- possibly using Producers + `ProducerIterator`